docs/specifications/tests/DNSSEC-TP/dnssec15.md

```diff
@@ -0,0 +1,173 @@
+DS15_INCONSISTENT_CDNSKEY | ERROR           | All servers do not have the same CDNSKEY RRset.
+DS15_INCONSISTENT_CDS     | ERROR           | All servers do not have the same CDS RRset.
+DS15_MISMATCH_CDS_CDNSKEY | ERROR           | Both CDS and CDNSKEY RRsets are found but they do not match.
+DS15_NO_CDS_CDNSKEY       | INFO            | No CDS or CDNSKEY RRsets are found on any name server.
+
+## Ordered description of steps to be taken to execute the test case
+
+1.  Create a CDS query with EDNS enabled with the DO bit set for the
+    apex of the *Child Zone*.
+
+2.  Create a CDNSKEY query with EDNS enabled with the DO bit set for
+    the apex of the *Child Zone*.
+
+4.  Retrieve all name server IP addresses for the *Child Zone* using
+    [Method4] and [Method5] ("NS IP").
+
+5.  Create the following empty sets:
+    1. Name server IP address and associated CDS RRset ("CDS RRsets"). A
+       name server IP can hold an empty RRset.
+    2. Name server IP address and associated CDNSKEY RRset ("CDNSKEY RRsets"). 
+       A name server IP can hold an empty RRset.
+    3. Name server IP set ("Mismatch CDS/CDNSKEY").
+
+6.  Repeat the following steps for each name server IP address in *NS IP*:
+
+    1. Send the CDS query over UDP to the name server IP address.
+       1. If no DNS response is returned, then go to next name server IP.
+       2. Else, if AA bit is not set in the DNS response, then go to next 
+          name server IP.
+       3. Else, if the RCODE in the DNS response is not *NOERROR*, then go to
+          next name server IP.
+       4. Else, if the DNS response contains no CDS record in the
+          answer section, then add the name server IP and an empty RRset to
+          the *CDS RRsets* set.
+       5. Else, add the name server IP and the CDS RRset from the answer
+          section to the *CDS RRsets* set.
+    2. Send the CDNSKEY query over UDP to the name server IP address.
+       1. If no DNS response is returned, then go to next name server IP.
+       2. Else, if AA bit is not set in the DNS response, then go to next
+          name server IP.
+       3. Else, if the RCODE in the DNS response is not *NOERROR*, then go to
+          next name server IP.
+       4. Else, if the DNS response contains no CDNSKEY record in the
+          answer section, then add the name server IP and an empty RRset to
+          the *CDNSKEY RRsets* set and go to next name server IP.
+       5. Else, add the name server IP and the CDNSKEY RRset from the answer
+          section to the *CDNSKEY RRsets* set and go to next name server IP.
+
+7.  If the two sets, *CDS RRsets* and *CDNSKEY RRsets*, both have only
+    empty RRsets then output *[DS15_NO_CDS_CDNSKEY]* and terminate this
+    test case.
+
+8.  If the *CDS RRsets* set has at least one non-empty RRset but no
+    non-empty RRsets in the *CDNSKEY RRsets* set then output
+    *[DS15_HAS_CDS_NO_CDNSKEY]*.
```

Comment:
`If [...] has at least one non-empty RRset but no non-empty RRsets in the [...] set`
->
`If [...] has at least one non-empty RRset but there are no non-empty RRsets in the [...] set`
Right?

The next item is like this too.

Reply:
Will fix through a larger rework.